### DISCUSSION TITLE
How to use transactions with Supabase?

### DISCUSSION THREAD

**Discussion by ioucyf** [supabase/supabase#11501] [Questions] [answered]
Is it even possible to use transactions without writing raw SQL? 

I know some ORM support them, but does supabase support them out of the box?

**Comment by GaryAustin1** [comment, marked as answer]
You can use an API rpc call to a postgres function and all code in that function is run as a transaction.   It is still SQL or plpgsql code in the function, but you can still use the API to call it.

**Reply by ioucyf** [reply]
That's actually very smart for an alternative method! 

Thank you very much for the suggestion.
